pkg/render/manager/manager_route_config.go

```diff
@@ -302,6 +305,54 @@ func (builder *voltronRouteConfigBuilder) mountConfigMapReference(name, key stri
 	return fmt.Sprintf("%s/%s/%s", configMapFolder, name, key), nil
 }
 
+// Generate annotations takes the keys / values we need to add annotations for and ensures they're formatted correctly,
+// i.e. are lower case and within the perceptible length. If the keys are too long, they are trimmed to 63 characters.
+// If two trimmed keys result in a conflict, the last bytes of the conflicting key are replaced with an incremental number
+// such that the length still remains 63 characters. This process continues, incrementing the prefix, until a non-conflicting
+// key has been generated.
+//
+// The keys are sorted before we generate the annotations, ensuring that the conflict resolution logic is deterministic.
+func (builder *voltronRouteConfigBuilder) generateAnnotations() map[string]string {
+	const maxAnnotationLen = 63
+
+	var keys []string
+	for key, _ := range builder.toAnnotate {
+		keys = append(keys, key)
+	}
+
+	sort.Strings(keys)
+	annotations := map[string]string{}
+	for _, key := range keys {
+		formattedKey := fmt.Sprintf("%s-%s", routesAnnotationPrefix, strings.ToLower(key))
+		if len(formattedKey) > maxAnnotationLen {
+			formattedKey = formattedKey[0:maxAnnotationLen]
+		}
+
+		// Just in case there's a collision
+		for i := 1; ; i++ {
+			if _, ok := annotations[formattedKey]; !ok {
+				break
+			}
+			suffix := strconv.Itoa(i)
+			formattedKey = formattedKey[0:len(formattedKey)-len(suffix)] + suffix
+			l := len(formattedKey)
```

Comment:
This looks like maybe something for debugging. Is this needed (and the line below)?

Reply:
Oh shoot, yes that wasn't supposed to be committed.

Reply:
Just removed it.